src/main/java/com/github/kiulian/downloader/parser/ParserImpl.java

```diff
@@ -105,7 +137,21 @@ private VideoInfo parseVideoAndroid(String videoId, YoutubeCallback<VideoInfo> c
             try {
                 formats = parseFormats(playerResponse, null, clientVersion);
 
-            } catch (YoutubeException e) {
+            }catch (InvalidJsUrlException e){
+                JSONObject playerConfig = downloadPlayerConfig("HHbt6rPuC3M",callback);
+                String jsUrl;
+                try {
+                    jsUrl = extractor.extractJsUrlFromConfig(playerConfig, videoId);
+                } catch (YoutubeException ex) {
```

Comment:
why do we need here another catch? it will just trigger `callback.onError(ex);` twice with same exception

Reply:
@iexavl what about this? seems like unnecessary call `callback.onError(ex);`